{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3d2ac7ee-02e3-4743-aa99-471da6458fda",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "import os\n",
    "import pandas as pd\n",
    "if not Path('./jw_utils').exists():\n",
    "    !git clone https://github.com/JonWinkelman/jw_utils.git\n",
    "if not Path('orthofinder_utils').exists():\n",
    "    !git clone https://github.com/JonWinkelman/orthofinder_utils.git\n",
    "from jw_utils import parse_fasta as pfa\n",
    "from jw_utils import hmmer_utils as hu\n",
    "from jw_utils import alignment_utils2 as au\n",
    "from jw_utils import itol_annotations as ia\n",
    "from jw_utils import ncbi_utils as nu\n",
    "\n",
    "from Bio import Phylo\n",
    "from Bio import SeqIO\n",
    "from plotly.subplots import make_subplots\n",
    "from plotly import graph_objects as go\n",
    "import plotly.io as pio\n",
    "#import make_queries as mq\n",
    "import random\n",
    "\n",
    "import subprocess\n",
    "from concurrent.futures import ThreadPoolExecutor\n",
    "from jw_utils import make_bac120_tree as mbt\n",
    "from orthofinder_utils import dash_ortho_parser_d as dop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d76eaa9-fd9b-4180-92d5-63256a34cd16",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7e0d85bf-1951-4807-ab16-b657e49e858d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Paths \n",
    "RESULTS = Path('./results')\n",
    "DATA=Path('./data')\n",
    "ASSETS=Path('./assets')\n",
    "BAC120_PROFILES = DATA / 'bac120_hmm_profiles'\n",
    "PROTEOME_DIR = DATA / 'Proteomes'\n",
    "REFERENCE_PROTEOME_PATH = PROTEOME_DIR / 'GCA_000014625.1.faa'\n",
    "ITOL_ANNOTATION_DIR = RESULTS / 'trees_itol_annotations'\n",
    "for dir in [RESULTS,DATA,ASSETS, ITOL_ANNOTATION_DIR]:\n",
    "    dir = Path(dir)\n",
    "    dir.mkdir(exist_ok=True, parents=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2140e449-48b1-44d4-8e63-c74aa8c9adda",
   "metadata": {},
   "source": [
    "### Build bac120 species tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6c3f5618-da8a-4135-b1c2-23ec71c33317",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-04-14 15:26:36,355 - INFO - Concatenating HMM profiles from data/bac120_hmm_profiles to /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm\n",
      "2025-04-14 15:26:36,460 - INFO - Running command: hmmpress /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "concatenating HMM profiles into one file and then pressing it into binary...\n",
      "All .HMM files from 'data/bac120_hmm_profiles' concatenated into '/Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm'.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-04-14 15:26:36,911 - INFO - hmmpress output:\n",
      "Working...    done.\n",
      "Pressed and indexed 120 HMMs (120 names and 120 accessions).\n",
      "Models pressed into binary file:   /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm.h3m\n",
      "SSI index for binary model file:   /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm.h3i\n",
      "Profiles (MSV part) pressed into:  /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm.h3f\n",
      "Profiles (remainder) pressed into: /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120hmm_db/concat_bac120_profiles.hmm.h3p\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "searching each proteome with each HMM using hmmsearch...\n",
      "aggregating best hit from each proteome into dataframe...\n",
      "Extracting protein sequences for each HMM from each proteome and writing FASTA files...\n",
      "5.15% finished.\n",
      "10.3% finished.\n",
      "15.15% finished.\n",
      "20.3% finished.\n",
      "25.15% finished.\n",
      "30.3% finished.\n",
      "35.15% finished.\n",
      "40.0% finished.\n",
      "45.15% finished.\n",
      "50.0% finished.\n",
      "55.15% finished.\n",
      "60.3% finished.\n",
      "65.15% finished.\n",
      "70.3% finished.\n",
      "75.15% finished.\n",
      "80.3% finished.\n",
      "85.15% finished.\n",
      "90.3% finished.\n",
      "95.15% finished.\n",
      "for each hmm, aligning all extracted proteins with that hmm\n",
      "Aligning PF01025.23...\n",
      "Aligning TIGR01171.1...\n",
      "Aligning TIGR01029.1...\n",
      "Aligning TIGR01953.1...\n",
      "Aligning TIGR02386.1...\n",
      "Aligning TIGR00459.1...\n",
      "Aligning TIGR00088.1...\n",
      "Aligning TIGR03632.1...\n",
      "Aligning TIGR01032.1...\n",
      "Aligning TIGR01951.1...\n",
      "Aligning TIGR00029.2...\n",
      "Aligning TIGR00392.1...\n",
      "Aligning TIGR00442.1...\n",
      "Aligning TIGR00810.1...\n",
      "Aligning TIGR00168.2...\n",
      "Aligning TIGR00663.1...\n",
      "Aligning TIGR00396.1...\n",
      "Aligning TIGR03594.1...\n",
      "Aligning TIGR00643.1...\n",
      "Aligning TIGR02729.1...\n",
      "Aligning TIGR00337.1...\n",
      "Aligning TIGR00116.1...\n",
      "Aligning TIGR00460.1...\n",
      "Aligning TIGR01009.1...\n",
      "Aligning TIGR00580.1...\n",
      "Aligning TIGR00250.1...\n",
      "Aligning PF00410.23...\n",
      "Aligning TIGR01071.1...\n",
      "Aligning TIGR01394.1...\n",
      "Aligning TIGR00095.1...\n",
      "Aligning PF00380.23...\n",
      "Aligning TIGR00928.1...\n",
      "Aligning TIGR01017.1...\n",
      "Aligning TIGR01393.1...\n",
      "Aligning TIGR00092.1...\n",
      "Aligning TIGR00090.1...\n",
      "Aligning TIGR01391.1...\n",
      "Aligning PF02576.21...\n",
      "Aligning TIGR02027.1...\n",
      "Aligning TIGR01169.1...\n",
      "Aligning TIGR00194.1...\n",
      "Aligning TIGR00420.1...\n",
      "Aligning TIGR01128.1...\n",
      "Aligning TIGR00115.1...\n",
      "Aligning TIGR03263.1...\n",
      "Aligning TIGR00539.1...\n",
      "Aligning TIGR02432.1...\n",
      "Aligning PF03726.18...\n",
      "Aligning TIGR00445.1...\n",
      "Aligning TIGR01011.1...\n",
      "Aligning TIGR00054.1...\n",
      "Aligning TIGR00487.1...\n",
      "Aligning TIGR03654.1...\n",
      "Aligning TIGR00083.1...\n",
      "Aligning TIGR00382.1...\n",
      "Aligning TIGR02191.1...\n",
      "Aligning TIGR00006.1...\n",
      "Aligning TIGR00020.2...\n",
      "Aligning TIGR00431.1...\n",
      "Aligning TIGR00065.1...\n",
      "Aligning TIGR00043.2...\n",
      "Aligning TIGR02012.1...\n",
      "Aligning TIGR00634.1...\n",
      "Aligning TIGR01039.1...\n",
      "Aligning TIGR00967.1...\n",
      "Aligning TIGR00922.1...\n",
      "Aligning TIGR00344.1...\n",
      "Aligning TIGR00158.1...\n",
      "Aligning TIGR00963.1...\n",
      "Aligning TIGR00755.1...\n",
      "Aligning TIGR00019.1...\n",
      "Aligning TIGR00435.1...\n",
      "Aligning TIGR00061.1...\n",
      "Aligning TIGR02350.1...\n",
      "Aligning TIGR01063.1...\n",
      "Aligning TIGR00362.1...\n",
      "Aligning TIGR00472.1...\n",
      "Aligning TIGR03953.1...\n",
      "Aligning TIGR01087.1...\n",
      "Aligning TIGR00456.1...\n",
      "Aligning TIGR02075.1...\n",
      "Aligning TIGR00496.1...\n",
      "Aligning TIGR01632.1...\n",
      "Aligning TIGR00064.1...\n",
      "Aligning TIGR01021.1...\n",
      "Aligning TIGR00398.1...\n",
      "Aligning TIGR02013.1...\n",
      "Aligning TIGR01510.1...\n",
      "Aligning TIGR02397.1...\n",
      "Aligning TIGR00059.1...\n",
      "Aligning TIGR01059.1...\n",
      "Aligning TIGR00414.1...\n",
      "Aligning TIGR00635.1...\n",
      "Aligning TIGR00959.1...\n",
      "Aligning TIGR00964.1...\n",
      "Aligning TIGR03725.1...\n",
      "Aligning TIGR01082.1...\n",
      "Aligning TIGR00416.1...\n",
      "Aligning TIGR00082.1...\n",
      "Aligning TIGR00717.1...\n",
      "Aligning TIGR01146.1...\n",
      "Aligning PF00466.24...\n",
      "Aligning TIGR01066.1...\n",
      "Aligning TIGR00186.1...\n",
      "Aligning TIGR00084.2...\n",
      "Aligning TIGR00436.1...\n",
      "Aligning TIGR00593.1...\n",
      "Aligning TIGR03625.1...\n",
      "Aligning TIGR01302.1...\n",
      "Aligning TIGR00086.1...\n",
      "Aligning TIGR00166.1...\n",
      "Aligning TIGR03723.1...\n",
      "Aligning TIGR01079.1...\n",
      "Aligning TIGR01164.1...\n",
      "Aligning TIGR00631.1...\n",
      "Aligning TIGR01044.1...\n",
      "Aligning TIGR00468.1...\n",
      "Aligning TIGR00138.1...\n",
      "Aligning TIGR00615.1...\n",
      "Aligning TIGR02273.1...\n",
      "For each proteome, concatenating all hmm-aligned proteins into one long protein and saving as results/bac120_tree/concatenated_alignment.fasta\n",
      "generating maximum likelihood tree using fasttree from concatenated protein sequences...\n",
      "FastTree successfully generated: /Users/jonathanwinkelman/Trestle_long-term-storage/Mukherjee_lab/mapping_PA14_20480_to_Pseudomonas/results/bac120_tree/bac120_tree.nwk\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PF01025.23</th>\n",
       "      <th>TIGR01171.1</th>\n",
       "      <th>TIGR01029.1</th>\n",
       "      <th>TIGR01953.1</th>\n",
       "      <th>TIGR02386.1</th>\n",
       "      <th>TIGR00459.1</th>\n",
       "      <th>TIGR00088.1</th>\n",
       "      <th>TIGR03632.1</th>\n",
       "      <th>TIGR01032.1</th>\n",
       "      <th>TIGR01951.1</th>\n",
       "      <th>...</th>\n",
       "      <th>TIGR00166.1</th>\n",
       "      <th>TIGR03723.1</th>\n",
       "      <th>TIGR01079.1</th>\n",
       "      <th>TIGR01164.1</th>\n",
       "      <th>TIGR00631.1</th>\n",
       "      <th>TIGR01044.1</th>\n",
       "      <th>TIGR00468.1</th>\n",
       "      <th>TIGR00138.1</th>\n",
       "      <th>TIGR00615.1</th>\n",
       "      <th>TIGR02273.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_001721805.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000759535.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_019334485.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900475285.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014268595.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_007833495.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014522265.1</th>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_009740185.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014076455.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_021495345.1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>330 rows Ã— 120 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 PF01025.23  TIGR01171.1  TIGR01029.1  TIGR01953.1  \\\n",
       "GCF_001721805.1        True         True         True         True   \n",
       "GCF_000759535.1        True         True         True         True   \n",
       "GCF_019334485.1        True         True         True         True   \n",
       "GCF_900475285.1        True         True         True         True   \n",
       "GCF_014268595.1        True         True         True         True   \n",
       "...                     ...          ...          ...          ...   \n",
       "GCF_007833495.1        True         True         True         True   \n",
       "GCF_014522265.1       False         True         True         True   \n",
       "GCF_009740185.1        True         True         True         True   \n",
       "GCF_014076455.1        True         True         True         True   \n",
       "GCF_021495345.1        True         True         True         True   \n",
       "\n",
       "                 TIGR02386.1  TIGR00459.1  TIGR00088.1  TIGR03632.1  \\\n",
       "GCF_001721805.1         True         True         True         True   \n",
       "GCF_000759535.1         True         True         True         True   \n",
       "GCF_019334485.1         True         True        False         True   \n",
       "GCF_900475285.1         True         True         True         True   \n",
       "GCF_014268595.1         True         True         True         True   \n",
       "...                      ...          ...          ...          ...   \n",
       "GCF_007833495.1         True         True         True         True   \n",
       "GCF_014522265.1         True         True        False         True   \n",
       "GCF_009740185.1         True         True         True         True   \n",
       "GCF_014076455.1         True         True         True         True   \n",
       "GCF_021495345.1         True         True         True         True   \n",
       "\n",
       "                 TIGR01032.1  TIGR01951.1  ...  TIGR00166.1  TIGR03723.1  \\\n",
       "GCF_001721805.1         True         True  ...         True         True   \n",
       "GCF_000759535.1         True         True  ...         True         True   \n",
       "GCF_019334485.1         True         True  ...         True         True   \n",
       "GCF_900475285.1         True         True  ...         True         True   \n",
       "GCF_014268595.1         True         True  ...         True         True   \n",
       "...                      ...          ...  ...          ...          ...   \n",
       "GCF_007833495.1         True         True  ...         True         True   \n",
       "GCF_014522265.1         True         True  ...         True         True   \n",
       "GCF_009740185.1         True         True  ...         True         True   \n",
       "GCF_014076455.1         True         True  ...         True         True   \n",
       "GCF_021495345.1         True         True  ...         True         True   \n",
       "\n",
       "                 TIGR01079.1  TIGR01164.1  TIGR00631.1  TIGR01044.1  \\\n",
       "GCF_001721805.1         True         True         True         True   \n",
       "GCF_000759535.1         True         True         True         True   \n",
       "GCF_019334485.1         True         True         True         True   \n",
       "GCF_900475285.1         True         True         True         True   \n",
       "GCF_014268595.1         True         True         True         True   \n",
       "...                      ...          ...          ...          ...   \n",
       "GCF_007833495.1         True         True         True         True   \n",
       "GCF_014522265.1         True         True         True         True   \n",
       "GCF_009740185.1         True         True         True         True   \n",
       "GCF_014076455.1         True         True         True         True   \n",
       "GCF_021495345.1         True         True         True         True   \n",
       "\n",
       "                 TIGR00468.1  TIGR00138.1  TIGR00615.1  TIGR02273.1  \n",
       "GCF_001721805.1         True         True         True         True  \n",
       "GCF_000759535.1         True         True         True         True  \n",
       "GCF_019334485.1         True         True         True         True  \n",
       "GCF_900475285.1         True         True         True         True  \n",
       "GCF_014268595.1         True         True         True         True  \n",
       "...                      ...          ...          ...          ...  \n",
       "GCF_007833495.1         True         True         True         True  \n",
       "GCF_014522265.1         True         True         True         True  \n",
       "GCF_009740185.1         True         True         True         True  \n",
       "GCF_014076455.1         True         True         True         True  \n",
       "GCF_021495345.1         True         True         True         True  \n",
       "\n",
       "[330 rows x 120 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ab_pres_df = mbt.make_bac120_tree(BAC120_PROFILES, \n",
    "                                  PROTEOME_DIR, \n",
    "                                  proteome_suffix='.faa', \n",
    "                                  output_tree='results/bac120_tree/bac120_tree.nwk',)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cddb0c6-dc7a-4fb0-9902-31b5cc48429d",
   "metadata": {},
   "source": [
    "#### make itol annotation file to relable species tree names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3b502a1e-0ec5-4b9a-a1c5-3485c7d0aa8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### make itol annotation file\n",
    "df = nu.make_summary_df('./data/summaries.json')\n",
    "acc_2_name_d = df['organism_name'].to_dict()\n",
    "relable_dict = {}\n",
    "for acc, org_name in acc_2_name_d.items():\n",
    "    new_name = f\" {'_'.join(org_name.split(' ')[:2])}_{acc}\"\n",
    "    relable_dict[acc] = new_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "70ec338f-6d93-46d1-bc77-ee2f14a4f4ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "BAC120_TREE_FP = Path('./results/bac120_tree/bac120_tree.nwk')\n",
    "bac120_tree = Phylo.read(str(BAC120_TREE_FP), format='newick')\n",
    "itol_relable_species_tree_fp = ITOL_ANNOTATION_DIR / 'itol_relable_species_tree.txt'\n",
    "ia.relabel_itol_treeleafs(bac120_tree, relable_dict, itol_relable_species_tree_fp)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83fdc083-53f6-4995-9377-ca8473dd87a6",
   "metadata": {},
   "source": [
    "### make itol annotation file to label presence/absence of bphP, algB, algW putative orthologs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "dccf2c5a-e1c1-42fd-a111-4e9f347a150a",
   "metadata": {},
   "outputs": [],
   "source": [
    "REFERENCE_PROTEOME_ACC='GCA_000014625.1'\n",
    "df_fp = 'data/GCA_000014625.1__v__GCF_000014625.1_full.csv'\n",
    "gene_prot_d = pd.read_csv(df_fp).set_index('GCA_gene_ID')['GCA_protein_ID'].to_dict()\n",
    "query_gene_id_d = {'DimA':'PA14_20480','bphP':'PA14_10700', 'algB':'PA14_72380', 'algW':'PA14_57760'}\n",
    "name_prot_d = {name:gene_prot_d[gene_id] for name, gene_id in query_gene_id_d.items()}\n",
    "\n",
    "color_d = {'bphP':'#05BE78','algB':'#C06000', 'algW':'#FF991C'}\n",
    "rb_result_df_d = {}\n",
    "for name, prot_id in name_prot_d.items():\n",
    "    if name != 'DimA':\n",
    "        rbresults_path = f'./results/rblast_results_{REFERENCE_PROTEOME_ACC}_{prot_id}/recip_blast_results_orgnames.csv'\n",
    "        rb_result_df = pd.read_csv(rbresults_path).set_index('target_proteome')\n",
    "        rb_result_df_d[name] = rb_result_df\n",
    "    \n",
    "        data_lst_of_ds = []\n",
    "        for acc, row in rb_result_df.iterrows():\n",
    "            if row['recip_best_hit']:\n",
    "                data_lst_of_ds.append({'node': acc, 'label': f'{name}_present', 'color': color_d[name], 'shape': '1'})\n",
    "            if not row['recip_best_hit']:\n",
    "                data_lst_of_ds.append({'node': acc, 'label': f'{name}_absent', 'color': '#D7D5D2', 'shape': '1'})\n",
    "                \n",
    "        colorstrip_itol_fp = ITOL_ANNOTATION_DIR / f'itol_colorstrip_{name}_presence.txt'\n",
    "        ia.make_itol_colorstrip_dataset(colorstrip_itol_fp, data_lst_of_ds, dataset_label=f'{name}_presence',\n",
    "                                     strip_width=50,legend_title=f'{name}_presence', color_branches=0)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da079a05-7917-4be0-8aa2-faf0872543bb",
   "metadata": {},
   "source": [
    "### make itol annotation file to label P. aeruginosa either clinical or environmental"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5f55ca23-9916-4978-b8a8-22498e32e183",
   "metadata": {},
   "outputs": [],
   "source": [
    "from jw_utils import plotly_preferences as pp\n",
    "data_lst_of_ds = []\n",
    "metadata_df = pd.read_csv('data//PA_strains.csv', index_col=0)\n",
    "for acc,row in metadata_df.iterrows():\n",
    "    if row['Type']=='clinical':\n",
    "        data_lst_of_ds.append({'node': acc, 'label': 'clinical', 'color': '#FF0000', 'shape': '1'})\n",
    "    else:\n",
    "        data_lst_of_ds.append({'node': acc, 'label': 'environmental', 'color': '#A0A0A0', 'shape': '1'})\n",
    "        \n",
    "clinical_environmental_colorstrip_itol_fp = ITOL_ANNOTATION_DIR / 'itol_colorstrip_clinical-vs-environmental.txt'\n",
    "ia.make_itol_colorstrip_dataset(clinical_environmental_colorstrip_itol_fp, data_lst_of_ds, dataset_label='clinical-vs-environmental',\n",
    "                             strip_width=50,legend_title='clinical-vs-environmental', color_branches=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc640ed6-4251-40b0-8a6e-1f248fb332d7",
   "metadata": {},
   "source": [
    "### Make colorstrip itol annotation for psuedomonas genera of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "baffad9e-6960-479d-89f9-f8662603ac6b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "genera_of_interest = ['chlororaphis','stutzeri', 'fluorescens', 'syringae', 'putida','aeruginosa']\n",
    "itol_colorstrip_list2 = []\n",
    "genera_of_interest = ['Pseudomonas', 'Metapseudomonas', 'Halopseudomonas', 'Geopseudomonas',\n",
    "                      'Ectopseudomonas', 'Phytopseudomonas', 'Aquipseudomonas', 'Candidatus', 'Stutzerimonas']\n",
    "for acc, org_name in acc_2_name_d.items():\n",
    "    genus = org_name.split(' ')[0]\n",
    "    species = ' '.join(org_name.split(' ')[:2])\n",
    "    if species == 'Pseudomonas aeruginosa':\n",
    "        itol_colorstrip_list2.append({'node':acc, 'label':genus, 'color':'#330000', 'shape':'1'})\n",
    "    elif genus in genera_of_interest:\n",
    "        itol_colorstrip_list2.append({'node':acc, 'label':genus, 'color':'#3399FF', 'shape':'1'})\n",
    "    else:\n",
    "        itol_colorstrip_list2.append({'node':acc, 'label':genus, 'color':'#C0C0C0', 'shape':'1'})\n",
    "\n",
    "\n",
    "itol_colorstrip_fp = 'results/trees_itol_annotations/itol_genus_colorstrip.txt'\n",
    "ia.make_itol_colorstrip_dataset(itol_colorstrip_fp, itol_colorstrip_list2, dataset_label='Psuedomonas_genera', legend_title='Psuedomonas_genera', \n",
    "                               strip_width=50, color_branches=1)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f3849e4f-41f2-4ce9-9693-40f26d76667e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dimA_rblast_df = pd.read_csv('results/rblast_results_GCA_000014625.1_ABJ12623.1/recip_blast_results_orgnames.csv', index_col=0)\n",
    "\n",
    "# add manual hits to dict  -- Manually detected dimA\n",
    "acc_prot_id_d = {\n",
    "                \"GCF_024168455.1\"  : \"WP_253559225.1\", \n",
    "                \"GCF_029350995.1\"  : \"WP_077934638.1\", \n",
    "                \"GCF_023241665.1\"  : \"WP_268261674.1\", \n",
    "                \"GCA_902497625.1\"  : \"WP_150579783.1\", \n",
    "                \"GCF_028657095.1\"  : \"WP_273924074.1\", \n",
    "                \"GCF_044900625.1\"  : \"WP_404697551.1\", \n",
    "                \"GCF_045062265.1\"  : \"WP_087273078.1\", \n",
    "                \"GCF_001698815.1\"  : \"WP_065832684.1\", \n",
    "                \"GCF_014357575.1\"  : \"WP_187522871.1\", \n",
    "                \"GCF_022790535.1\" : \"WP_243246698.1\", \n",
    "                \"GCF_900103875.1\" : \"WP_090187081.1\",\n",
    "                } \n",
    "# add rblast hits to dict\n",
    "for acc, row in dimA_rblast_df.iterrows():\n",
    "    if row['recip_best_hit']:\n",
    "        acc_prot_id_d[acc] = row['f_bbhit']\n",
    "\n",
    "data_lst_of_ds = []\n",
    "all_dimA_hits = list(acc_prot_id_d.keys())\n",
    "for acc, row in dimA_rblast_df.iterrows():\n",
    "    if acc in all_dimA_hits:\n",
    "        data_lst_of_ds.append({'node': acc, 'label': 'present', 'color': '#AD07E3', 'shape': '1'})\n",
    "    else:\n",
    "        data_lst_of_ds.append({'node': acc, 'label': 'present', 'color': '#D7D5D2', 'shape': '1'})\n",
    "        \n",
    "dimA_colorstrip_itol_fp = ITOL_ANNOTATION_DIR / 'itol_colorstrip_DimA.txt'\n",
    "ia.make_itol_colorstrip_dataset(dimA_colorstrip_itol_fp, data_lst_of_ds, dataset_label='dimA',\n",
    "                             strip_width=50,legend_title='dimA', color_branches=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0601e3fe-4e7b-4a06-8621-b3af6894063f",
   "metadata": {},
   "source": [
    "### make itol dataset to annotate trees with absence/presence of WTF MOTIF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "11e6df72-6269-41c1-b6bf-84c0e6c8e2b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_prot_seqs(proteome_dir, acc_prot_id_d):\n",
    "    \"\"\"\"\"\"\n",
    "    seq_d = {}\n",
    "    for acc, prot_id  in acc_prot_id_d.items():\n",
    "        fp = Path(proteome_dir) / f'{acc}.faa'\n",
    "        f_seq_d = pfa.get_seq_dict(fp)\n",
    "        seq = f_seq_d[prot_id]\n",
    "        full_id = f'{acc}_{prot_id}'\n",
    "        seq_d[full_id] = seq\n",
    "    return seq_d\n",
    "\n",
    "seq_d  = extract_prot_seqs(PROTEOME_DIR, acc_prot_id_d)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d454ed4b-5ec5-4e05-9b5f-8f3acbd21a32",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "pattern = r\"[WL][TVI]F$\"\n",
    "motif_d = {}\n",
    "for full_id, seq in seq_d.items():\n",
    "    m = re.search(pattern, seq[-3:]) \n",
    "    if m:\n",
    "        motif_d[full_id[:15]] = [True, m[0]]\n",
    "    else:\n",
    "        motif_d[full_id[:15]] = [False, '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "acc830f7-a315-484f-962e-c8fa30cba8fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_lst_of_ds = []\n",
    "for acc, d_lst  in motif_d.items():\n",
    "    if d_lst[0]:\n",
    "        data_lst_of_ds.append({'node': acc, 'label': 'present', 'color': '#B6202F', 'shape': '2'})\n",
    "    else:\n",
    "        data_lst_of_ds.append({'node': acc, 'label': 'absent', 'color': '#E2EDF3', 'shape': '2'})\n",
    "        \n",
    "motif_colorstrip_itol_fp = ITOL_ANNOTATION_DIR / 'itol_colorstrip_motif.txt'\n",
    "ia.make_itol_colorstrip_dataset(motif_colorstrip_itol_fp, data_lst_of_ds, dataset_label='motif_present',\n",
    "                             strip_width=50,legend_title='motif_present', color_branches=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5176e73b-27f3-4486-81de-1d976c845257",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e0b83d0-cdfe-4856-b666-ec68557ee65d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de8b0226-e212-42a3-b109-290dd9ff957e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
